Realm/RLMAccessor.hpp

```diff
@@ -0,0 +1,116 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMAccessor.h"
+
+#import "object_accessor.hpp"
+
+#import "RLMUtil.hpp"
+
+@class RLMRealm;
+class RLMClassInfo;
+class RLMObservationInfo;
+
+// realm::util::Optional<id> doesn't work due to that explicitly invoking the
+// destructor isn't allowed under ARC, so this covers the subset of Optional
```

Comment:
Isn't the immediate problem that Obj-C types cannot be members of a union under ARC? Explicitly invoking the destructor of a variable of type `id` appears to be allowed, but doesn't do anything useful since the destructor isn't what handles releasing the object that the variable points to.